docs/Features.md

```diff
@@ -0,0 +1,216 @@
+style, macro invocation is the same.
+
+By default, macro invocations start with a long arrow `==>` and end with a period `.`. Both of these
+sequences can be customized to suit your needs better. The only restriction with macro invocations
+is that they must be the only thing on the line. That is, this is valid:
+
+```rs
+fn main() {
+  ==> Calculate the very complex result.
+  ==> Print the results for the user.
+}
+```
+
+But this would not invoke the macro named within the `if()` as the macro sequences do not start and
+end the line:
+
+```rs
+fn main() {
+  if (==> A very complex condition is true.) {
+    ==> Do something cool.
+  }
+}
+```
+
+Another feature of macros to note is that if two code blocks have the same name, they are
+concatenated, in the order they are written. This can be very useful in defining global variables or
+listing imports closer to the parts where they are used.
+
+## Meta variables
+
+If you consider a macro invocation like a function call, then meta variables are like parameters.
+
+By default, to indicate that a macro includes a meta variable, the name of the variable must be part
+of the name of the macro, delimited by `@{` and `}`.
+
+Then that meta variable may be used within the macro by again using its name within the `@{` and `}`
+in the code.
+
+Finally, a macro with meta variables is invoked by replacing the name of the variable with its
+value in the invocation.
+
+An example:
+
+~~~md
+Here is our macro with meta variables:
+
+```rust - Say @{something} to @{someone}
+println!("Hey, @{someone}! I was told to tell you \"@{something}\"");
+```
+
+Now, to say things to many people:
+
+```rust
+==> Say @{Hello} to @{Jim}.
+==> Say @{How are you} to @{Tom}.
+==> Say @{I am good!} to @{Angela}.
+```
+~~~
+
+Meta variables can have default values:
+
+~~~md
+Here is our macro with default meta variables:
+
+```rust - Say @{something:Hello} to @{someone}
+println!("Hey, @{someone}! I was told to tell you \"@{something}\"");
+```
+
+Now, to say the default "Hello" to many people:
+
+```rust
+==> Say @{} to @{Jim}.
+==> Say @{} to @{Tom}.
+==> Say @{} to @{Angela}.
+```
+~~~
+
+These features allows for more flexibility when writing macros, as well as possibly making the intent
+clearer.
+
+## Extracted comments
+
+By default, the comment extraction sequence is set to `//`, purely for familiarity. Any text after
+(and including) this sequence is extracted from the code block, and not rendered to the tangled
+source code. Note that, since the comments are removed completely when compiling, they do not have
+to use the actual line comment indicator from you programming language. In fact, it may be better to
+choose a sequence that is *not* the regular comment indicator so that you can still have comments in
+your tangled code output.
+
+Now that these comments are extracted, it is possible to handle them differently in the weaved
```

Comment:
Also here, the question seems to be why not mark these comments up in the actual document because that is where the human-readable presentation is supposed to happen.

Reply:
I think removing "normal" (language) comments from the source is too harsh.

In addition, comments in the output may become useful once there is a "reverse mode". 

And again, why should humans not **want** to read the code? Maybe to check if everything works as expected. Or because they are fed up with LP, and want to continue "the normal way".

Reply:
Note that here I am referring to the Yarner-specific comments, not the target language comments. Those are put into markup-specific tags and such and this is what I am questioning: If I want to have an `<aside>` here, why not just put that into the surrounding markup instead of relying on the extraction of Yarner-specific comments?

Reply:
Because this may split up code too much, or because you may somehow want to "label" particular parts of the code that you describe above or below the block?

Reply:
The `<aside>` will be an aside whether its markup is before, after or within the code block. The Yarner-specific comments will not be visible in the source code anyway. And the important thing here is the layout of the human-readable document, not the layout of its mark-up, which will be the same in both approaches. (Consider someone using e.g. LyX to edit the LaTeX document, he/she will never actually see that mark-up.) Also I cannot label stuff using comments that will be removed, right?

This is about the Yarner-specific comments that will not end up in the source code at all. I just think that "extract the Yarner-specific comments into Yarner-specific mark-up in the target document files and strip it from the target source files" is too complicated and too fringe a feature to be useful especially as the same effect can be achieved by marking up the asides explicitly with exactly the same result in the generated human-readable document.